test/services/NotProcessedFilesServiceSpec.scala

```diff
@@ -0,0 +1,230 @@
+import models.notification.{RecordStatusEnum, SDESAudit, SDESChecksum, SDESNotification, SDESNotificationFile, SDESProperties}
+import org.mockito.Matchers
+import org.mockito.Mockito.{mock, reset, times, verify, when}
+import org.scalatest.concurrent.Eventually.eventually
+import play.api.Configuration
+import play.api.http.Status.INTERNAL_SERVER_ERROR
+import play.api.test.Helpers.{NO_CONTENT, await, defaultAwaitTimeout}
+import repositories.FileNotificationRepository
+import scheduler.ScheduleStatus
+import uk.gov.hmrc.http.HttpResponse
+import uk.gov.hmrc.mongo.lock.MongoLockRepository
+import utils.Logger.logger
+import utils.PagerDutyHelper.PagerDutyKeys
+import utils.{LogCapturing, TimeMachine}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration.{Duration, DurationInt}
+
+class NotProcessedFilesServiceSpec extends SpecBase with LogCapturing {
+  val mockLockRepository: MongoLockRepository = mock(classOf[MongoLockRepository])
+  val mockSDESConnector: SDESConnector = mock(classOf[SDESConnector])
+  val mockConfig: Configuration = mock(classOf[Configuration])
+  val mockTimeMachine: TimeMachine = mock(classOf[TimeMachine])
+  val mockFileNotificationRepository: FileNotificationRepository = mock(classOf[FileNotificationRepository])
+  val jobName = "NotProcessedFilesService"
+
+  val mongoLockId: String = s"schedules.$jobName"
+  val mongoLockTimeout: Int = 123
+  val releaseDuration: Duration = mongoLockTimeout.seconds
+
+  val mockDateTime: LocalDateTime = LocalDateTime.of(2022, 1, 1, 0, 0, 0)
+  val notification: SDESNotification = SDESNotification(
+    informationType = "info",
+    file = SDESNotificationFile(
+      recipientOrSender = "penalties",
+      name = "file1.txt",
+      location = "http://example.com",
+      checksum = SDESChecksum(algorithm = "SHA-256", value = "123456789-abcdef-123456789"),
+      size = 256,
+      properties = Seq.empty[SDESProperties]
+    ),
+    audit = SDESAudit("file 1")
+  )
+
+
+  val notificationRecord: SDESNotificationRecord = SDESNotificationRecord(
+    reference = "ref1",
+    status = RecordStatusEnum.PENDING,
+    numberOfAttempts = 1,
+    createdAt = mockDateTime.minusHours(1),
+    updatedAt = mockDateTime,
+    nextAttemptAt = mockDateTime,
+    notification = notification
+  )
+
+  val pendingNotifications: Seq[SDESNotificationRecord] = Seq(
+    notificationRecord,
+    notificationRecord.copy(reference= "ref2", nextAttemptAt = mockDateTime.minusSeconds(1), status = RecordStatusEnum.FILE_RECEIVED_IN_SDES),
+    notificationRecord.copy(reference = "ref3", nextAttemptAt = mockDateTime.plusHours(1), status = RecordStatusEnum.FILE_RECEIVED_IN_SDES)
+  )
+
+  val pendingNotificationTwo = Seq(
+    notificationRecord,
+    notificationRecord.copy(reference= "ref2", nextAttemptAt = mockDateTime.minusSeconds(1)),
+    notificationRecord.copy(reference= "ref3", nextAttemptAt = mockDateTime.minusSeconds(3))
+  )
+
```

Comment:
What is the difference between this and `pendingNotifications` other than the nextAttemptAt?